src/main/java/core/basesyntax/SalaryInfo.java

```diff
@@ -1,7 +1,53 @@
 package core.basesyntax;
 
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
 public class SalaryInfo {
     public String getSalaryInfo(String[] names, String[] data, String dateFrom, String dateTo) {
-        return null;
+        DateTimeFormatter formatter = DateTimeFormatter.ofPattern("dd.MM.yyyy");
+        LocalDate startDate;
+        LocalDate endDate;
+
+        try {
+            startDate = LocalDate.parse(dateFrom.trim(), formatter);
+            endDate = LocalDate.parse(dateTo.trim(), formatter);
+        } catch (DateTimeParseException e) {
+            throw new IllegalArgumentException("Invalid date format. "
+                    + "Expected format: dd.MM.yyyy", e);
+        }
+
+        int[] salaries = new int[data.length];
```

Comment:
The 'salaries' array should be the same length as the 'names' array, not the 'data' array. Each employee's salary should correspond to their name, not to each data record.